compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -203,37 +230,46 @@ generateSrcFromLf env = noLoc mod
         IEModuleContents noExt (noLoc ghcModName)
 
     classReexports :: [Gen (LIE GhcPs)]
-    classReexports = map snd (MS.toList classReexportMap)
+    classReexports = map snd (MS.elems classReexportMap)
 
-    classReexportMap :: MS.Map LF.TypeSynName (Gen (LIE GhcPs))
+    classReexportMap :: MS.Map LF.TypeSynName (LF.PackageId, Gen (LIE GhcPs))
     classReexportMap = MS.fromList $ do
         synDef@LF.DefTypeSyn{..} <- NM.toList . LF.moduleSynonyms $ envMod env
-        LF.TStruct _ <- [synType]
+        guard $ case synType of
+            LF.TStruct _ -> True
+            LF.TUnit -> True
```

Comment:
It might be good to add a small comment explaining when this happens.

Reply:
Good idea, I added a comment.